### PR TITLE
Password UI component

### PR DESCRIPTION
**Pull request by Reshex**
Added Password UI example in the chat area for you to see

### CODE REVIEW

**Review by talyaron — COMMENTED** [review]
Hi Bar, The password has to be above all other components, and also when pressing the first number and then the secons number on the keybaord, the numbers the cursor has to move to the next input field automaticly. Please do not PR, untill you are connected with Vladi's code.

**Comment by Reshex** [comment]
> Hi Bar, The password has to be above all other components, and also when pressing the first number and then the secons number on the keybaord, the numbers the cursor has to move to the next input field automaticly. Please do not PR, untill you are connected with Vladi's code.

Hey Tal thank you for the feedback.
Maybe you checked on the hebrew version, the cursor is moving as expected in the english version, I wanted to ask you if you want me to make the hebrew version also. The password above all that you mentioned I didn't quite understood, do you mean z index?

File: src/view/components/passwordUi/passwordUi.module.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      background-color: var(--button-blue);
+    }
+  }
+
+  .passwordUi__buttonSection {
+    margin-top: 6rem;
+
+    button {
+      justify-content: center;
+      width: 15rem;
```
Comment (vladbyk89) [inline src/view/components/passwordUi/passwordUi.module.scss:67]: Try not to use fixed sizes for elements. 
Look into using % and vh/vw

File: src/view/components/passwordUi/PasswordUi.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+				text: t(`Something went wrong. Please try again!`),
+				textStyle: styles.passwordUi__statusSection__passwordTextIncorrect
+			});
+
+			setTriesCounter(prev => {
+				const newTriesCounter = prev + 1;
+				if (newTriesCounter >= MAX_TRIES) {
+					navigate("/401");
+				}
+				return newTriesCounter;
```
Comment (vladbyk89) [inline src/view/components/passwordUi/PasswordUi.tsx:64]: Expected newline before return statement.

File: src/view/components/passwordUi/PasswordUi.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	function handleSubmit(event: MouseEvent<HTMLButtonElement>) {
+		event.preventDefault();
+
+		const enteredCode = Number(values.join(""));
+
+		if (enteredCode === PASSWORD_CODE) {
+			setPasswordCheck(true);
+			setPasswordState({
+				img: passwordUiImgGreen,
+				text: t(`Bravo! Your passcode is correct. Welcome aboard!`),
```
Comment (vladbyk89) [inline src/view/components/passwordUi/PasswordUi.tsx:49]: passcode? think of a different word. Perhaps a Password or Pin?

**Review by vladbyk89 — CHANGES_REQUESTED** [review]
Very good job on the functionality of the component. 
I added a few UI comments.
Think about the case when a user creates a statement. Why would he have to type in the code.

File: src/view/components/passwordUi/passwordUi.module.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
+@import url("https://fonts.googleapis.com/css2?family=Bellefair&display=swap");
+
+.passwordUi {
+  margin-top: 5rem; // temporary
+  margin-bottom: 5rem; // temporary
+  margin-left: 50rem;
```
[the remark this answers is not in the document]
Comment (Reshex) [reply src/view/components/passwordUi/passwordUi.module.scss:6]: Yes, I used it in order to test something and forgot it was there, i will fix it and center the component

File: src/view/pages/statement/components/chat/StatementChat.tsx [hunk 5]
```diff
@@ -76,37 +79,46 @@ const StatementChat: FC<Props> = ({
...
+							newMessages={newMessages}
+							setNewMessages={setNewMessages}
+							scrollToBottom={scrollToBottom}
 						/>
+						{statement && (
+							<StatementInput
+								toggleAskNotifications={toggleAskNotifications}
+								statement={statement}
+							/>
+						)}
```
[the remark this answers is not in the document]
Comment (Reshex) [reply src/view/pages/statement/components/chat/StatementChat.tsx:111]: I will think about it

[the remark this answers is not in the document]
Comment (Reshex) [reply src/view/pages/statement/components/chat/StatementChat.tsx:111]: I have thought about doing that if the user id = to the creator id than the password check state will be true. But it means that the creator of the statement will never have to input the password. Is that a good idea? also how can i check if that works in real time?


//
useEffect(() => {
		if (user?.uid === statement.creatorId) {
			setPasswordCheck(true)
		}
		else{
			setPasswordCheck(false)
		}
	}, [])
	
	//

[the remark this answers is not in the document]
Comment (vladbyk89) [reply src/view/pages/statement/components/chat/StatementChat.tsx:111]: This sounds like a good idea. The statement creator also generates the password. If he can't log in then this statement will be locked forever. 

What do you mean by checking in real-time? 

You can create a user and statement when running the emulator, correct? Then use the same link to open the statement in another window (incognito will be best) and check what the "normal" user flow looks like.

[the remark this answers is not in the document]
Comment (Reshex) [reply src/view/pages/statement/components/chat/StatementChat.tsx:111]: > This sounds like a good idea. The statement creator also generates the password. If he can't log in then this statement will be locked forever.
> 
> What do you mean by checking in real-time?
> 
> You can create a user and statement when running the emulator, correct? Then use the same link to open the statement in another window (incognito will be best) and check what the "normal" user flow looks like.

I have tried to do as you said and it seems to work as expected, ill push the changes for you to see.
Thank you very much :)

**Comment by vladbyk89** [comment]
Can you resolve conflicts and fix the type check error so I can merge it?

**Comment by vladbyk89** [comment]
Great job so far! I appreciate the work you've done.

I have just one more suggestion regarding the UI component: When I input a code, I currently have to click the button to submit it. Could you add functionality so that pressing "Enter" also submits the code?

Additionally, could you display the remaining number of attempts left for the user?

**Comment by Reshex** [comment]
> Great job so far! I appreciate the work you've done.
> 
> Just a few suggestions regarding the UI component:
> 
> * When I input a code, I must click the button to submit it. Could you add functionality so that pressing "Enter" also submits the code?
> * Could you display the remaining number of attempts left for the user?
> 
> Additionally, I noticed that the PasswordUi component should be moved to StatementMain. Otherwise, the password flow won't work if the user tries to load the solutions page, for example.

At first I thought doing it exactly like that but in figma it was like what I have done. I will update it according to your requests

**Pull request merged**